exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -280,8 +281,15 @@ public void interrupted(final InterruptedException ex) {
     }
   }
 
-  QueryState updateEphemeralState(final QueryState queryState) {
-    switch (queryState) {
+  void updateEphemeralState(final QueryState queryState) {
+      // If query is already in zk transient store, ignore the transient state update option.
+      // Else, they will not be removed from transient store upon completion.
+      if (transientProfiles.get(stringQueryId) == null &&
```

Comment:
I want to bypass the option for the queries which are already in transient store when option is enabled. Otherwise, their state will never get updated and/or will never be removed from transient store. web UI will show these queries as running forever :-)

Thanks for raising a good point regarding using transientProfiles.get. I made the change to update and use in memory state instead. 

Please review the new diffs.